src/apps/main/switcher.cpp

```diff
@@ -43,9 +43,10 @@ void OswAppSwitcher::loop() {
     if (hal->btnHasGoneUp(_btn)) {
         if (_doSleep) {
             _doSleep = false;
+            _doSwitch = false; // if lightsleep is enabled then doSwitch is still true and will be used the next time.
+            sleeping = true; // because in lightsleep the next _app will use the button and switch the app which isn't what we want 
```

Comment:
Not declared, and it should prevent initialization when it wake up.

Reply:
Whoops I forgot to add the sleeping variable initialization as I worked with two different forks and just drag and dropped my changes.